packages/grid/x-data-grid-pro/src/hooks/features/detailPanel/gridDetailPanelToggleColDef.tsx

```diff
@@ -9,7 +9,7 @@ export const GRID_DETAIL_PANEL_TOGGLE_FIELD = '__detail_panel_toggle__';
 export const GRID_DETAIL_PANEL_TOGGLE_COL_DEF: GridColDef = {
   ...GRID_STRING_COL_DEF,
   field: GRID_DETAIL_PANEL_TOGGLE_FIELD,
-  type: 'detailPanelToggle',
+  type: 'string',
```

Comment:
Does it make sense to have a type dedicated to all the interactive columns (all the toggles, the actions, etc...)
Because "string" is not super coherent IMHO

Reply:
They're extending the `string` column definition anyway, it's just that they don't render the string as cell content 🙂
Adding another type would make it appear in the autocomplete for the users, I would rather not add it until there are more reasons to do so.

Reply:
But regardless if we add a different `"custom"` type, we should probably do something like this for our custom columns:

![image](https://github.com/mui/mui-x/assets/1423607/b708d1cd-2861-4cb7-814a-ec29c08f02b9)

Reply:
> I think we've already debated on it, but I'll repeat anyway here for the record: I think we should add a "custom" type for the columns that don't fit with any real datatype. It's more explicit than having our special columns be "string"

Yes that's basically what I had in mind

Reply:
I've added the `custom` column type, but did not create `GRID_CUSTOM_COL_DEF`:

- Every custom column type has a different set of flags and there's no general rule for that: some columns have the column menu, some columns have row grouping disabled, etc. It's better to have all the flags explicitly provided in each column definition.
- Not all custom columns extend the string column definition - see https://github.com/mui/mui-x/blob/8dff180a89d9cc6e01634e23176abf78637b71a3/packages/grid/x-data-grid/src/colDef/gridCheckboxSelectionColDef.tsx#L10-L12

What do you think?